lib/services/databaseobjectimportrule.go

```diff
@@ -0,0 +1,97 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package services
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	dbobjectimportrulev1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/dbobjectimportrule/v1"
+)
+
+// DatabaseObjectImportRule is the DatabaseObjectImportRule service
+type DatabaseObjectImportRule interface {
+	// CreateDatabaseObjectImportRule will create a new DatabaseObjectImportRule resource.
+	CreateDatabaseObjectImportRule(ctx context.Context, rule *dbobjectimportrulev1.DatabaseObjectImportRule) (*dbobjectimportrulev1.DatabaseObjectImportRule, error)
+
+	// UpsertDatabaseObjectImportRule creates a new DatabaseObjectImportRule or forcefully updates an existing DatabaseObjectImportRule.
+	UpsertDatabaseObjectImportRule(ctx context.Context, rule *dbobjectimportrulev1.DatabaseObjectImportRule) error
+
+	// GetDatabaseObjectImportRule will get a DatabaseObjectImportRule resource by name.
+	GetDatabaseObjectImportRule(ctx context.Context, name string) (*dbobjectimportrulev1.DatabaseObjectImportRule, error)
+
+	// DeleteDatabaseObjectImportRule will delete a DatabaseObjectImportRule resource.
+	DeleteDatabaseObjectImportRule(ctx context.Context, name string) error
+
+	// UpdateDatabaseObjectImportRule updates an existing DatabaseObjectImportRule.
+	UpdateDatabaseObjectImportRule(ctx context.Context, rule *dbobjectimportrulev1.DatabaseObjectImportRule) (*dbobjectimportrulev1.DatabaseObjectImportRule, error)
+
+	// ListDatabaseObjectImportRules will list DatabaseObjectImportRule resources.
+	ListDatabaseObjectImportRules(ctx context.Context, size int, token string) ([]*dbobjectimportrulev1.DatabaseObjectImportRule, string, error)
```

Comment:
What does the token do?

Reply:
`token` is a standard argument for `ListFoo` RPCs; it tells you where to resume the listing.

Reply:
I've renamed it `pageToken` to convey additional context, I hope this works well enough.